packages/core/FieldElement.h

```diff
@@ -73,7 +73,7 @@ class FieldElement: public Field
          *--------------------------------------------------------------------*/
 
         operator bool() const {
-            return value != 0;  // Object is "true" if value is non-zero
+            return static_cast<T>(value) != 0; // Object is "true" if value is non-zero
```

Comment:
I am very nervous about this change.  The code, as it was, will allow the compiler to issue an error when it is inappropriately used.  I think the new code would do the same, but I am not sure why there would ever be a reason to perform a static cast of a variable of type T to the exact same type that it already is.

The problem with this change, is that I know the original code will cause the compiler to issue an error when I do something that doesn't make sense, but I don't know how the new code will behave.